### PR TITLE
Contributed upgrades

### PR DESCRIPTION
**Pull request by hrstoyanov**
- Remove unnecessary dependencies, makes dokan-java standalone!
- Upgrade packages
- Minor fixes to build system, code.

### CODE REVIEW

**Comment by infeo** [comment]
I'll put the review on hold for a week, because maybe this project will change the default branch to develop. And if that happens, please submit your PR to `develop`.

File: gradle.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name=dokan-java
-version=1.1.0
+version=1.2.0
```
Comment (infeo) [inline gradle.properties]: As long as no new version is released, a `-SNAPSHOT` suffix should be added to the version (if possible).

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,6 @@
+#Wed Jan 22 15:25:48 PST 2020
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.1-all.zip
```
Comment (infeo) [inline gradle/wrapper/gradle-wrapper.properties]: are there any breaking changes with gradle 6?

File: src/main/java/dev/dokan/dokan_java/structure/filesecurity/SecurityIdentifier.java [hunk 3]
```diff
@@ -16,7 +14,7 @@
  */
 public class SecurityIdentifier implements Byteable {
 
-	private static final Logger LOG = LoggerFactory.getLogger(SecurityIdentifier.class);
+//	private static final Logger LOG = LoggerFactory.getLogger(SecurityIdentifier.class);
```
Comment (infeo) [inline src/main/java/dev/dokan/dokan_java/structure/filesecurity/SecurityIdentifier.java]: This line can be removed

File: src/main/java/dev/dokan/dokan_java/structure/filesecurity/SecurityIdentifier.java [hunk 4]
```diff
@@ -38,7 +36,7 @@ public SecurityIdentifier(SidIdentifierAuthority sidAuth, List<Integer> subAutho
 			if (subAuthorities.size() <= 15) {
 				this.subAuthorities.addAll(subAuthorities);
 			} else {
-				LOG.warn("Number of subauthorities exceeds the limit of 15. Only taking the first 15 ones.");
+				//LOG.warn("Number of subauthorities exceeds the limit of 15. Only taking the first 15 ones.");
```
Comment (infeo) [inline src/main/java/dev/dokan/dokan_java/structure/filesecurity/SecurityIdentifier.java]: In my opinion there _should_ be a warning. Maybe we need to throw an exception here.

File: src/test/java/dev/dokan/dokan_java/examples/DirListingFSTest.java [hunk 5]
```diff
@@ -4,24 +4,14 @@
...
 public class DirListingFSTest {
 
-    public static final Logger LOG = LoggerFactory.getLogger(DirListingFSTest.class);
-
-    static {
-        System.setProperty(SimpleLogger.DEFAULT_LOG_LEVEL_KEY, "info");
-        System.setProperty(SimpleLogger.LOG_FILE_KEY, "System.out");
-        System.setProperty(SimpleLogger.SHOW_DATE_TIME_KEY, "true");
-        System.setProperty(SimpleLogger.DATE_TIME_FORMAT_KEY, "HH:mm:ss:SSS");
-    }
```
Comment (infeo) [inline src/test/java/dev/dokan/dokan_java/examples/DirListingFSTest.java]: Why removing the logger dependecy? This is an example and with logging the user can see what happen. I don't think it it should be removed.

File: src/main/java/dev/dokan/dokan_java/examples/DirListingFileSystem.java [hunk 6]
```diff
@@ -242,7 +241,7 @@ public int getVolumeInformation(Pointer rawVolumeNameBuffer, int rawVolumeNameSi
 
     private Path getrootedPath(WString rawPath) {
         String unixPath = rawPath.toString().replace('\\', '/');
-        String relativeUnixPath = CharMatcher.is('/').trimLeadingFrom(unixPath);
+        String relativeUnixPath = unixPath.startsWith("/")?unixPath.substring(1):unixPath;
```
Comment (infeo) [inline src/main/java/dev/dokan/dokan_java/examples/DirListingFileSystem.java]: This is not equivalent:
For a path which only contains `/`, the first one returns an empty string while the latter one throws an exception.

File: dokan-java.iml [hunk 7]
```diff
@@ -1,13 +0,0 @@
...
-    <exclude-output />
-    <content url="file://$MODULE_DIR$">
-      <excludeFolder url="file://$MODULE_DIR$/.gradle" />
-      <excludeFolder url="file://$MODULE_DIR$/build" />
-      <excludeFolder url="file://$MODULE_DIR$/out" />
-    </content>
-    <orderEntry type="inheritedJdk" />
-    <orderEntry type="sourceFolder" forTests="false" />
-  </component>
-</module>
```
Comment (infeo) [inline dokan-java.iml]: Should be added to gitignore then

File: src/main/java/dev/dokan/dokan_java/structure/filesecurity/SecurityIdentifier.java (hunk 4 above)
Comment (hrstoyanov) [reply src/main/java/dev/dokan/dokan_java/structure/filesecurity/SecurityIdentifier.java]: Ok. This is the only reason to have dependency to Sl4J, maybe we can do something else without requireing SLF4J?

File: src/test/java/dev/dokan/dokan_java/examples/DirListingFSTest.java (hunk 5 above)
Comment (hrstoyanov) [reply src/test/java/dev/dokan/dokan_java/examples/DirListingFSTest.java]: We can keep the SLF4J for all tests, no problem with that. I should probably correct this

File: src/main/java/dev/dokan/dokan_java/examples/DirListingFileSystem.java (hunk 6 above)
Comment (hrstoyanov) [reply src/main/java/dev/dokan/dokan_java/examples/DirListingFileSystem.java]: I see ... but just a simple think like this brings the entire Guava dependency! Guava is a very questionable library  for  modern Java ( it was good until Java 7)

File: dokan-java.iml (hunk 7 above)
Comment (hrstoyanov) [reply dokan-java.iml]: Agreed. The github sources should not depend on any particular IDE (I too, use Intellij)

File: build.gradle [hunk 8]
```diff
@@ -13,19 +13,19 @@ plugins {
     id 'com.jfrog.bintray' version '1.8.4'
 }
 
+tasks.withType(JavaCompile) {
+    options.incremental = true
+    options.encoding = 'UTF-8'
+    sourceCompatibility = JavaVersion.VERSION_11
+    targetCompatibility = JavaVersion.VERSION_11
+    options.compilerArgs << "-Xlint:unchecked" << "-Xlint:deprecation" << "-Xlint:deprecation"
```
[the remark this answers is not in the document]
Comment (hrstoyanov) [reply build.gradle]: my bad, should be once only.

[the remark this answers is not in the document]
Comment (hrstoyanov) [reply build.gradle]: btw, when you enable extra compiler checks, some warning were reported in th etests. Have not had time to look into them ... will do.

File: gradle.properties (hunk 1 above)
Comment (hrstoyanov) [reply gradle.properties]: I can make is SNAPSHOT, no prolem. Btw, I saw "1.1.0" somewhere in the java code, hardcoded ...

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (hrstoyanov) [reply gradle/wrapper/gradle-wrapper.properties]: There are deprecated APIs in Gradle 6, for example in publishing the "classifier" attributed had to be changed.

File: src/main/java/dev/dokan/dokan_java/structure/filesecurity/SecurityIdentifier.java (hunk 3 above)
Comment (hrstoyanov) [reply src/main/java/dev/dokan/dokan_java/structure/filesecurity/SecurityIdentifier.java]: ok

**Comment by hrstoyanov** [comment]
@infeo 
Thanks for reviewing the merge request! I will submit a new one against the branch you want. In general, the goal is:
- strengthen the compilation flags
- upgrade libs 
- try to remove dependencies (unless they bring significant value, like JNA at the moment)
- update Gradle (Gradle moves too fast with too many changes)

File: src/main/java/dev/dokan/dokan_java/examples/DirListingFileSystem.java (hunk 6 above)
Comment (hrstoyanov) [reply src/main/java/dev/dokan/dokan_java/examples/DirListingFileSystem.java]: btw, maybe the entire examples/ folder need to be move to tests?

**Pull request closed without merging**